### PR TITLE
Vue Migration

### PR DESCRIPTION
**Pull request by jasonadt**
## Description
This is our big VueJS migration for the UI libs (including Storybook for documentation)

You can already play with Storybook (which was recently updated to have the same bits as what's in this PR) here: https://lana.github.io/b2c-mapp-ui/

## Testing

- `npm run lint`: **PASS**
- `npm run test`: **PASS**
- `npm run storybook`: **PASS**

## Checks
- [x] Requires documentation update
- [x] Requires lib version update

## Versioning impact
- [x] **MAJOR** incompatible API changes.
- [ ] **MINOR** adds functionality in a backwards-compatible manner.
- [ ] **PATCH** backwards-compatible bug fixes.

### CODE REVIEW

**Comment by jasonadt** [comment]
A special big THANK YOU to @jantocv for porting over our UI Library Unit Tests to VueJS! 🎉 🥂 🙇‍♂️

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name: Build
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    env:
+      working-dir: ./
+      TOKEN: ${{ secrets.GITHUB_ACCESS_TOKEN }}
```
Comment (flimzy) [inline .github/workflows/nodejs.yml:19]:
```suggestion
      TOKEN: ${{ secrets.GITHUBACCESSTOKEN }}
```

File: .github/workflows/mergepr.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    env:
+      working-dir: ./
+      TOKEN: ${{ secrets.GITHUB_ACCESS_TOKEN }}
```
Comment (flimzy) [inline .github/workflows/mergepr.yml:21]: This is unfortunately inconsistent across repos, making this a very easy mistake to make.

```suggestion
      TOKEN: ${{ secrets.GITHUBACCESSTOKEN }}
```

**Comment by jantocv** [comment]
> A special big THANK YOU to @jantocv for porting over our UI Library Unit Tests to VueJS! 🎉 🥂 🙇‍♂️

**Pull request merged**